src/main/perf/LineFileDocs.java

```diff
@@ -156,7 +163,7 @@ private void readDocs() throws Exception {
           throw new RuntimeException("expected " + length + " document bytes but read " + x);
         }
         buffer.position(0);
-        queue.put(buffer);
+        queue.put(new LineFileDoc(buffer, readVector()));
```

Comment:
Hmm this is a bit spooky to me -- the reason that we were passing `byte[]` previously was to have each indexing thread handle the `byte[]` to `Document` conversion, to minimize the (bottleneck'd) single-threaded "pull documents from a single file" here.  I think that bottleneck mattered in the past (indexing throughput was hurt on many-cored boxes).

Could we just make a simple `Pair` class to hold the `byte[]` buffer and `float[]` vector, and then keep the `LineFileDocs` parsing/creation multi-threaded?

Reply:
This LineFileDoc class *is* what you are asking for, I think? I'm not sure why it appears to be parsing? I just added it as a cleanup task so that we can have a single type handling both byte[] and String cases which were a bit messy before, with casts from Object. We did add some cost here reading the vector, but I think that is extremely fast, analogous to reading the document text as we already do.

Reply:
D'oh!  You are right, that class is just a thin wrapper on top of the otherwise unparsed line -- sorry, I was confused.  Looks great!